src/rws_client.cpp

```diff
@@ -103,6 +103,17 @@ void RWSClient::SubscriptionResources::add(const std::string& resource_uri, cons
  * Primary methods
  */
 
+RWSClient::RWSResult RWSClient::getContollerService()
+{
+  std::string uri = Services::CTRL;
+
+  EvaluationConditions evaluation_conditions;
+  evaluation_conditions.parse_message_into_xml = true;
+  evaluation_conditions.accepted_outcomes.push_back(HTTPResponse::HTTP_OK);
+
+  return evaluatePOCOResult(httpGet(uri), evaluation_conditions);
```

Comment:
If any of this fails, will it throw exceptions, or silently fail?

`httpGet(..)` fi?

Reply:
`httpGet(-..)` can fail and throw exceptions, but they are are caught in the method and reported in its return value. A failure could for example be communication timeout.

`evaluatePOCOResult(...)` will see the failure status and report it in its return value.